base/docs/Docs.jl

```diff
@@ -297,22 +350,23 @@ function docm(meta, def, define = true)
         # `@doc "<doc string>" :(str_macro"")` syntax.
         #
         # Allow more general macrocall for now unless it causes confusion.
-        return objdoc(meta, namify(def′.args[1]))
+        return vardoc(meta, nothing, namify(def′.args[1]))
     end
 
     define || (def = nothing)
 
     fexpr(def′)                && return funcdoc(meta, def, def′, namify(def′))
+    isexpr(def′, :function)    && return namedoc(meta, def, namify(def′))
     isexpr(def′, :call)        && return funcdoc(meta, nothing, def′, namify(def′))
     isexpr(def′, :type)        && return typedoc(meta, def, namify(def′.args[2]))
-    isexpr(def′, :macro)       && return namedoc(meta, def, symbol("@", namify(def′)))
+    isexpr(def′, :macro)       && return  vardoc(meta, def, symbol("@", namify(def′)))
     isexpr(def′, :abstract)    && return namedoc(meta, def, namify(def′))
     isexpr(def′, :bitstype)    && return namedoc(meta, def, def′.args[2])
+    isexpr(def′, :typealias)   && return  vardoc(meta, def, namify(def′))
     isexpr(def′, :module)      && return namedoc(meta, def, def′.args[2])
-    isexpr(def′, :(=), :const) && return namedoc(meta, def, namify(def′))
+    isexpr(def′, :(=), :const) && return  vardoc(meta, def, namify(def′))
 
-    isexpr(def′, :macrocall) && (def = namify(def′))
-    return objdoc(meta, def)
+    objdoc(meta, def′)
```

Comment:
As an example, documenting a module that is defined during bootstrap with

```
"..."
MyModule
```

takes that code path and we end up documenting ``nothing`` if ``def`` is used instead of ``def′``.